spec/models/service_plan_spec.rb

```diff
@@ -52,7 +52,29 @@
       end
 
       it "shows the modified column is unchanged" do
-        expect(service_plan.modified["schema"]).to eq JSON.parse(valid_ddf)["schema"]
+        expect(service_plan.modified["schema"]).to eq valid_ddf["schema"]
+      end
+    end
+
+    context "modified schema comparison" do
+      let(:service_plan) { create(:service_plan, :base => reordered_ddf) }
+      let(:reordered_ddf) do
+        valid_ddf.dup.tap do |ddf|
+          # pushing the first field onto the end of the array so they're in a different order.
+          field = ddf["schema"]["fields"].shift
+          ddf["schema"]["fields"] << field
```

Comment:
This is cute, but doesn't something simple like `valid_ddf["schema"]["fields"].reverse` work too?

Reply:
Ahhh yeah thats a good point. Added a `.reverse!` so it manipulates the array in place.